internal/bufferedwrites/buffered_write_handler.go

```diff
@@ -71,6 +71,14 @@ func (wh *BufferedWriteHandler) Write(data []byte, offset int64) (err error) {
 		return fmt.Errorf("non sequential writes")
 	}
 
+	// Fail early if the uploadHandler has failed.
+	select {
```

Comment:
please extract this into method as it is used at 2 places

Reply:
I'll take care of this in the sync/flush changes PR.